docs/proposals/30-apiserver-proxy.md

```diff
@@ -0,0 +1,268 @@
+- When [ACL extension](https://github.com/stackitcloud/gardener-extension-acl) is actively filtering traffic based on source IP addresses
+- When multiple shoot clusters are using overlapping private IP ranges
+- When the LoadBalancer is not maintaining the original client source IP information
+
+The issue does not impact environments where:
+- The LoadBalancer operates in transparent mode
+- Only a single proxy protocol header is present
+- Alternative authentication methods are the primary security control
+- Client IP ranges are strictly segregated between shoots
+
+![Current State with opaque LB](./assets/30-current-architecture.png "Current State with opaque LB")
+
+## Proposed Changes
+
+1. Custom Header Implementation
+    - Introduce a new custom header `X-Gardener-API-Route`
+    - Define the structure and encoding of the header value to include necessary routing information
+
+2. New Port/Path
+    - Implement a new port or path on the `istio-ingressgateway` for the new routing mechanism
+    - Ensure this new route is separate from any existing VPN or tunnel infrastructure
+
+3. HTTP CONNECT Reconfiguration:
+    - Reconfigure the `apiserver-proxy` to use HTTP CONNECT for establishing a secure tunnel
+    - This will replace the current TCP proxy protocol method
+    - The HTTP CONNECT method will be used on the newly implemented port/path
+
+4. Feature Gate
+    - Implement a feature gate to control the rollout of the new routing mechanism
+    - Define stages for the feature gate: alpha, beta, and stable
+
+5. Phased Implementation
+    - Phase 1: Add the new port/path and custom header processing
+    - Phase 2: Gradually reconfigure shoots to use the new routing method
+    - Phase 3: Deprecate and remove the old routing method
+
+![Proposed Architecture](./assets/30-proposed-architecture-http-connect.png "Proposed Architecture")
+![Proposed Architecture](./assets/30-proposed-architecture-tls.png "Proposed Architecture")
+    
+    
+## Custom Header Specification
+
+The new custom header will be structured as follows:
+
+```
+X-Gardener-API-Route: <routing-information>
+```
+
+The routing information will be a concatenated string, joined by pipes, containing the following fields:
+
+- `direction`: The direction of traffic.
+- `port`: The destination port.
+- `subset`: Istio subset/version.
+- `destination`: The destination.
+
+Putting this to practice should result in a header formatted like this:
+
+`X-Gardener-API-Route: outbound|8134||kube-apiserver.foobar.svc.cluster.local`
+
+During the transition period, the system should be able to handle both the new custom header and the existing routing method. This can be controlled via the feature gate.
+
+If the header is missing, malformed, or fails verification, the error should be logged with appropriate details for debugging and return an HTTP 400 (Bad Request) status code. It should never fall back to the old routing method to ensure security.
+
+## HTTP CONNECT Implementation
```

Comment:
Let's also link for HTTP Connect, e.g. https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.6.